code/game/machinery/camera/trackable.dm

```diff
@@ -82,47 +61,140 @@
 	var/list/targets = sort_list(humans) + sort_list(others)
 	return targets
 
-///Toggles whether or not we're tracking something. Arg is whether it's on or off.
-/datum/trackable/proc/set_tracking(on = FALSE)
-	if(on)
+/// Takes a mob to track, resets our state and begins trying to follow it
+/// Best we can at least
+/datum/trackable/proc/set_tracked_mob(mob/living/track)
+	set_rechecking(FALSE)
+	if(tracked_mob)
+		UnregisterSignal(tracked_mob, list(COMSIG_QDELETING, COMSIG_MOVABLE_MOVED, COMSIG_MOVABLE_UPDATE_GLIDE_SIZE))
+	if(track && !isliving(track))
+		tracked_mob = null
+		return
+	tracked_mob = track
+	if(tracked_mob)
```

Comment:
early return?

Reply:
IDK. I like this pattern for signal tied setters but I can't really describe why. I will if you want

Reply:
Probably okay then